pkg/resizer/csi_resizer.go

```diff
@@ -68,15 +70,18 @@ func NewResizerFromClient(
 	k8sClient kubernetes.Interface,
 	informerFactory informers.SharedInformerFactory,
 	metricsManager metrics.CSIMetricsManager,
+	metricsServer metrics.Server,
 	metricsAddress, metricsPath string) (Resizer, error) {
 	driverName, err := getDriverName(csiClient, timeout)
 	if err != nil {
 		return nil, fmt.Errorf("get driver name failed: %v", err)
 	}
 
 	klog.V(2).Infof("CSI driver name: %q", driverName)
-	metricsManager.SetDriverName(driverName)
-	metricsManager.StartMetricsEndpoint(metricsAddress, metricsPath)
+	if metricsAddress != "" {
```

Comment:
is it possible to move this outside so that it's similar to how csi-attacher does it?

Reply:
I added the refactor as a separate commit. WDYT?